src/python/gudhi/simplex_tree.py

```diff
@@ -0,0 +1,456 @@
+        Diagonal entries are currently interpreted as vertices, although we do not guarantee this behavior
+        in the future, and this is only useful if you want to insert vertices with a smaller filtration value
+        than the smallest edge containing it, since vertices are implicitly inserted together with the edges.
+
+        :param edges: the edges to insert and their filtration values.
+        :type edges: scipy.sparse.coo_matrix of shape (n,n)
+
+        .. seealso:: :func:`insert_batch`
+        """
+        # Without this, it could be slow if we end up inserting vertices in a bad order (flat_map).
+        super().insert_batch_vertices(
+            np.unique(np.stack((edges.row, edges.col))), float("inf")
+        )
+        # TODO: optimize this?
+        for edge in zip(edges.row, edges.col, edges.data):
+            super().insert((edge[0], edge[1]), edge[2])
+
+    def insert_batch(
+        self,
+        vertex_array: Union[np.ndarray[np.int32], np.ndarray[np.int64]],
+        filtrations: Union[np.ndarray[np.float32], np.ndarray[np.float64]],
+    ):
+        """Inserts k-simplices given by a sparse array in a format similar
+        to `torch.sparse <https://pytorch.org/docs/stable/sparse.html>`_.
+        The n-th simplex has vertices `vertex_array[0,n]`, ...,
+        `vertex_array[k,n]` and filtration value `filtrations[n]`.
+        If a simplex is repeated, the smallest filtration value is used.
+        Simplices with a repeated vertex are currently interpreted as lower
+        dimensional simplices, but we do not guarantee this behavior in the
+        future. Any time a simplex is inserted, its faces are inserted as well
+        if needed to preserve a simplicial complex.
+
+        :param vertex_array: the k-simplices to insert.
+        :type vertex_array: numpy.array of shape (k+1,n)
+        :param filtrations: the filtration values.
+        :type filtrations: numpy.array of shape (n,)
+        """
+        vertices = np.unique(vertex_array)
+        k = vertex_array.shape[0]
+        n = vertex_array.shape[1]
+        assert filtrations.shape[0] == n, "inconsistent sizes for vertex_array and filtrations"
+        v = []
+        # Without this, it could be slow if we end up inserting vertices in a bad order (flat_map).
+        # NaN currently does the wrong thing
+        super().insert_batch_vertices(vertices, float("inf"))
+        for i in range(n):
+            for j in range(k):
+                v.append(vertex_array[j, i])
+            super().insert(v, filtrations[i])
+            v.clear()
+
+    def get_simplices(self):
+        """This function returns a generator with simplices and their given
+        filtration values.
+
+        :returns:  The simplices.
+        :rtype:  generator with tuples(simplex, filtration)
+        """
+        for sh in super().simplex_iter():
+            yield super().get_simplex_and_filtration(sh)
```

Comment:
Isn't this code **significantly slower** than previous cython code ? The cython code was *compiling* this iterator. Now it's a python for loop... The same goes for the next iterators btw.


Also, I think this is natural. What do you think about this ?
```suggestion
    def __iter__(self):
        """This function returns a generator with simplices and their given
        filtration values.

        :returns:  The simplices.
        :rtype:  generator with tuples(simplex, filtration)
        """
        for sh in super().simplex_iter():
            yield super().get_simplex_and_filtration(sh)
     def get_simplices(self):
         from warnings import warn
         warn(
            "TODO.",
            DeprecationWarning,
        )
        for stuff in self:
            yield stuff
         
```

Reply:
For all the method of the form "for sh in some_c++_iterator", the solution would be to replace the `yield` loop simply with an iterator, i.e., instead of `simplex_iter` (direct c++ binding) returning a simplex handle, a method returning directly an iterator to the simplex+filtration pair. Otherwise, we could just explicitly construct the array.
@mglisse @VincentRouvreau what do you think?

Reply:
IIRC at the beginning we used to build a list eagerly, but switched to a lazy generator (`yield`) because the list could take more memory than the simplex tree itself (by a factor `dim`), which defeats the purpose of using a simplex tree.
I think this function is mostly for users who want to write a loop in Python (`for s,f in st.get_simplices():`), so optimizing it a bit may not give significant gains.
On the other hand, writing something that has the interface of a Python generator in C++/nanobind is easy (https://nanobind.readthedocs.io/en/latest/api_extra.html#iterator-bindings), so why not.